sharktank/sharktank/examples/export_paged_llm_v1.py

```diff
@@ -121,26 +126,32 @@ def generate_batch_prefill(bs: int):
             "tokens": {1: sl_dim},
             "seq_lens": {},
             "seq_block_ids": {1: block_dim},
-            "cache_state": cache_state_dynamic_shapes,
+            "cs": cache_state_dynamic_shapes * llama_config.tensor_parallelism_size,
         }
 
         print(f"Exporting prefill_bs{bs}")
 
         @fxb.export_program(
             name=f"prefill_bs{bs}",
-            args=(tokens, seq_lens, seq_block_ids, cache_state),
+            args=(
+                tokens,
+                seq_lens,
+                seq_block_ids,
+                [shard._data for shard in cache_state[0].shards],
```

Comment:
We have the sharded tensor types registered in PyTorch and can be a part of function signatures. They will be auto-flattened/unflattened during export.
Then you would not need to reconstruct the cache from the pieces latter.

Reply:
It is fine to leave it like that, not a big deal.

Reply:
I reworked the helpers for packing / unpacking. Ultimately we should writing some export tooling to automate this. Anything sharded should generate the affinities, etc.